src/cart/crt_rpc.c

```diff
@@ -1628,8 +1628,13 @@ crt_rpc_priv_init(struct crt_rpc_priv *rpc_priv, crt_context_t crt_ctx, bool srv
 
 	crt_rpc_inout_buff_init(rpc_priv);
 
-	rpc_priv->crp_timeout_sec = (ctx->cc_timeout_sec == 0 ? crt_gdata.cg_timeout :
-				     ctx->cc_timeout_sec);
+	if (srv_flag)
+		rpc_priv->crp_timeout_sec = rpc_priv->crp_req_hdr.cch_src_timeout == 0 ?
```

Comment:
on servers you should be using context timeout too if its set. the order of timeout precedence is (low to high) global, per-context, per-rpc 

i feel it would be simpler to keep the old logic lines 1631-1632 and just add extra check after such as

```
if (srv_flag && rpc_priv->crp_req_hdr.cch_src_timeout)
    rpc_priv->crp_timeout_sec = rpc_priv->crp_req_hdr.cch_src_timeout;
```

Reply:
fixed, thx

Reply:
hmm it shows no changes since last review, did you forget to push something?

Reply:
it added "rpc_priv->crp_req_hdr.cch_src_timeout == 0" checking for the old client case...